### PR TITLE
Store min stage

### PR DESCRIPTION
**Pull request by ghost**
Add the ability to store the minimum stage in SWALS.

### CODE REVIEW

**Comment by ghost** [comment]
TODO: check tests pass. Validation test BP09 explicitly uses this new flag

**Comment by ghost** [comment]
> TODO: check tests pass. Validation test BP09 explicitly uses this new flag

BP09 passes using ifort on NCI

**Comment by ghost** [comment]
Inspection of outputs from BP09 shows that the min stage is NaN (-3.3e38 to -9.9e30).

**Comment by ghost** [comment]
Initialised to +huge instead of -huge.
Inspection of min_stage results on example BP09 looks correct. 
Added tests to BP09 to ensure that the min_stage has the same min and max values on one domain in each future run.

File: propagation/SWALS/examples/nthmp/BP09/test_min_stage.R [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        multidomain_dir = most_recent_run, 
+        domain_index=domain_number,
+        desired_var = 'min_stage',
+        return_raster=TRUE
+    )
+    min_stage = as.matrix(min_stage_raster)
+
+    # check min minimum stage is -5.987428
+    check_min_stage = min(min_stage, na.rm=TRUE)
+    if (abs(min(check_min_stage) - -5.987428) > atol) {
```
Comment (gareth-d-ga) [inline propagation/SWALS/examples/nthmp/BP09/test_min_stage.R]: This test is very fragile -- any minor change to the solver will cause it to fail. Can you please make it less fragile (e.g. within a few percent is OK).

File: propagation/SWALS/src/util/netcdf_util_mod.f90 [hunk 2]
```diff
@@ -36,8 +36,8 @@ module netcdf_util
     character(len=charlen), parameter :: gridded_output_variables_time(4) = &
         [character(len=charlen) :: 'stage', 'uh', 'vh', 'elev']
         !! Gridded variables that can be output as time-series. Must match [STG, UH, VH, ELV] in order.
-    character(len=charlen), parameter :: gridded_output_variables_max_U(5) = &
-        [character(len=charlen) :: 'max_stage', 'max_speed', 'max_flux', 'arrival_time', 'time_of_max_stage']
+    character(len=charlen), parameter :: gridded_output_variables_max_U(6) = &
+        [character(len=charlen) :: 'max_stage', 'min_stage', 'max_speed', 'max_flux', 'arrival_time', 'time_of_max_stage']
```
Comment (gareth-d-ga) [inline propagation/SWALS/src/util/netcdf_util_mod.f90]: Could you please move `min_stage` to the end of this array? It probably doesn't matter. But the comment mentions assumptions that `max_stage` is first. I would not be shocked if somewhere there exists a model which assumes that `max_speed` is second. If we add new variables to the end, then problems seem less likely.

**Comment by gareth-d-ga** [comment]
Thanks Matt, this looks useful. Could you please make the updates above, and confirm that the unit tests and validation tests all pass on NCI? Then I'm happy to merge.

**Review by gareth-d-ga — CHANGES_REQUESTED** [review]

**Comment by gareth-d-ga** [comment]
I added a commit to the R test script to avoid halting on failure.

File: propagation/SWALS/examples/nthmp/BP09/test_min_stage.R (hunk 1 above)
Comment (ghost) [reply propagation/SWALS/examples/nthmp/BP09/test_min_stage.R]: Solved in [3c42c55](https://github.com/GeoscienceAustralia/ptha/pull/49/commits/3c42c55ca7e02277ca38b3c511d4a5304fc634e7) by using a relative tolerance of 1%

File: propagation/SWALS/src/util/netcdf_util_mod.f90 (hunk 2 above)
Comment (ghost) [reply propagation/SWALS/src/util/netcdf_util_mod.f90]: Resolved in [3c42c55](https://github.com/GeoscienceAustralia/ptha/pull/49/commits/3c42c55ca7e02277ca38b3c511d4a5304fc634e7)

**Comment by ghost** [comment]
All unit tests, parallel tests and validation tests pass on NCI using ifort.

**Pull request merged**